### PR TITLE
feat: lazy load js-ipfs-http-client and async iterator support

### LINKED ISSUE #6: Dynamically loaded ipfs-http-client
Dear Friends,
I'm working on a wiki dapp and needed to lighten its overall size.
I load ipfs-http-client only when users need to update their wiki.
I fork ipfs-provider and made the necessary updates:.
https://github.com/xmaysonnave/tiddlywiki-ipfs
is built against
https://github.com/xmaysonnave/ipfs-provider
 - travis runs correctly.
 - I added a .nvmrc as I use nvm.
 - I didn't mock the loader
Thanks

### PR DESCRIPTION
**Pull request by lidel**
This PR:

- removes `js-ipfs-http-client` from the default bundle, 
  (enabling people to lazily load it only when it is actually needed)
  - Supersedes #7, Closes #6 
  - We want to ship it together with  https://github.com/ipfs-shipyard/ipfs-provider/issues/3 
- adds support for async iterator JS APIs
  https://blog.ipfs.io/2020-02-01-async-await-refactor/ 
- **updates `README` and code in `./examples`**
  - **TLDR**: see updated README at https://github.com/ipfs-shipyard/ipfs-provider/blob/feat/lazy-load-ipfs-http-client/README.md
- ensure URL implementation is correct in Node environments (`iso-url`)

First two are breaking changes, so I plan to release as v1.0.0.
We need this for unblocking some collabs, such as kiwix-js.

### CODE REVIEW

**Comment by hacdias** [comment]
Are you gonna work on #3? If not, I can take a stab at it!

**Comment by lidel** [comment]
@hacdias I won't have bandwidth for it next week, so if you want, its yours :)

**Comment by lidel** [comment]
Notes to self: 
- this needs more time, so we won't release v1 yet, after #17 is merged we will do v0.5 and this will land in v1
- this PR needs to be rebased on top of #17
- would be good to figure out how to proivide "batteries included" experience (if user does not provide own URL)
- if we hardcode any URLs, those should be trustless (backed by SRI verification)
- any improvements done for lazy-loading http-client should be also applied to js-ipfs (this PR should close both #2 and #6)

**Comment by lidel** [comment]
@hacdias mind looking at this? I want to merge this and then release it as v1.0.0,
but it may impact your use cases, as you created #19  recently.

**Comment by xmaysonnave** [comment]
@lidel Thanks for the new build. Integrated and tested on my side. Lazy load without the internal dependency is working as suggested. Great
